File: iojs/lib/fizzy.js
#!/usr/bin/env iojs

"use strict";

function fizzBuzz(n) {
  var
  a = n % 3 === 0,
  b = n % 5 === 0;

  if (a && b) {
    return "FizzBuzz";
  }
  else if (a) {
    return "Fizz";
  }
  else if (b) {
    return "Buzz";
  }
  else {
    return n.toString();
  }
}

function main() {
  for (var i = 1; i < 101; i++) {
    console.log(fizzBuzz(i));
  }
}

if (!module.parent) { main(); }

File: perl6/fizzy.pl
#!/usr/bin/env perl6

sub fizzy {
  my $n = shift @_;

  my ($a, $b) = ($n % 3 == 0, $n % 5 == 0);

  if ($a && $b) {
    return "FizzBuzz";
  }
  elsif ($a) {
    return "Fizz";
  }
  elsif ($b) {
    return "Buzz";
  }
  else {
    return "$n";
  }
}

sub MAIN {
  my @fbs = 1 .. 100;

  for (0 .. 99) {
    @fbs[$_] = start {
      fizzy($_ + 1);
    }
  };

  for (@fbs) {
    my $r = $_.result;
    print "$r\n";
  }
}
